miso-web/src/main/webapp/scripts/hot_qctype.js

```diff
@@ -0,0 +1,89 @@
+HotTarget.qctype = {
+  createUrl: '/miso/rest/qctype',
+  updateUrl: '/miso/rest/qctype/',
+  requestConfiguration: function(config, callback) {
+    callback(config)
+  },
+  fixUp: function(qctype, errorHandler) {
+  },
+  createColumns: function(config, create, data) {
+    return [
+      HotUtils.makeColumnForText('Name', true, 'name', {
+        unpackAfterSave: true,
+        validator: HotUtils.validator.requiredText
+      }), HotUtils.makeColumnForText('Description', true, 'description', {
+        unpackAfterSave: true,
+      }), HotUtils.makeColumnForEnum('Target', create, true, 'qcTarget', Constants.qcTargets, null),
+      HotUtils.makeColumnForText('Units', true, 'units', {
+        unpackAfterSave: true,
+      }), {
+        header: 'Is Boolean',
+        data: 'isBoolean',
+        type: 'dropdown',
+        required: true,
+        trimDropdown: false,
+        source: ['False', 'True'],
+        include: true,
+        unpack: function(obj, flat, setCellMeta) {
+          flat['isBoolean'] = obj['precisionAfterDecimal'] === -1 ? 'True' : 'False';
+        },
+        pack: function(obj, flat, errorHandler) {
+          if (flat['isBoolean'] === 'True') {
+            obj['precisionAfterDecimal'] = -1;
+          } else {
+            obj['precisionAfterDecimal'] = null;
+          }
+        },
+        depends: 'precisionNumber',
+        update: function(obj, flat, flatProperty, value, setReadOnly, setOptions, setData) {
+          if(!Utils.validation.isEmpty(flat[flatProperty])){
+            setReadOnly(true);
+          } else {
+            setReadOnly(false)
+          }
+        }
+      }, {
+        header: 'Precision After Decimal',
+        data: 'precisionNumber',
+        type: 'numeric',
+        include: true,
+        validator: HotUtils.validator.requiredPositiveInt,
+        unpack: function(obj, flat, setCellMeta) {
+          flat['precisionNumber'] = Utils.valOrNull(obj['precisionAfterDecimal']);
+        },
+        pack: function(obj, flat, errorHandler) {
+          if (!Utils.validation.isEmpty(flat['precisionNumber'])) {
+            obj['precisionAfterDecimal'] = flat['precisionNumber'];
+          }
+        },
+        depends: 'isBoolean',
+        update: function(obj, flat, flatProperty, value, setReadOnly, setOptions, setData) {
+          if(value === 'True'){
+            setOptions({
+              'validator': HotUtils.validator.requiredEmpty
+            })
+            setData = '';
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "setData" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)